lib/medtronic/medtronicSimulator.js

```diff
@@ -66,6 +77,111 @@ exports.make = function(config){
   }
 
   return {
+    basal: function(event) {
+      ensureTimestamp(event);
+
+      if (currBasal != null) {
+
+        if(currBasal.deliveryType === 'temp' && event.deliveryType === 'temp' && event.duration !== 0) {
+          // temp basal was updated
+          currBasal.with_expectedDuration(currBasal.duration)
+                   .with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          event.suppressed = _.clone(currBasal.suppressed);
```

Comment:
On the pump you can actually edit a running temp basal's duration. As such, I'm stopping the old temp basal, starting a new one with the new duration, and copying over the suppressed info. Should this rather result in the entire edited temp basal being added to the suppressed info? It's these kind of things I'm a bit unclear about since the docs don't exist yet.